src/compile.rs

```diff
@@ -207,6 +270,17 @@ fn compile_target(
         // but forwarding stderr is still useful in case there some non-json error
         .stderr(Stdio::inherit());
 
+    if let Some(zig_linker) = zig_linker {
+        build_command.env("CC", &zig_linker);
```

Comment:
I think this should also use `format!("{}_CC", env_target)` instead of just `CC` because user might have already set that.